README.md
# ELS - Enterprise Linux/EuroLinux Sources - libvisio
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
2015-01-02  David Tardon  <dtardon@redhat.com>  [0478fce6799d9a8e2d765cc4616a5f2de1c3c44e]

	comment out 2 unused functions in test

	Change-Id: I7012a55176693b9abb8e9f5c9543d14221bea5e1

2015-01-02  David Tardon  <dtardon@redhat.com>  [0a11bbae2d3fbc834a08393ace6671bdf2909696]

	disable werror by default

	Change-Id: I389f93a756f3661b328847cde267f79cffefef62

2015-01-02  David Tardon  <dtardon@redhat.com>  [02eea2cf353bb405c5073d9169868a93b29f197c]

	prepare for a release

	Change-Id: I1022e4003f468d7c66eec459b41016ec1ab53556

2015-01-01  David Tardon  <dtardon@redhat.com>  [bec680b235f06b439340fbc25b20b2f5a131300f]

	Revert "VSDX: import <cp:category>"

	We do not want "multi-level" property names.

	This reverts commit 5753f7e98d02bde80d9e712d298372933c4a28e6.

2014-12-31  Miklos Vajna  <vmiklos@collabora.co.uk>  [5753f7e98d02bde80d9e712d298372933c4a28e6]

	VSDX: import <cp:category>

	Change-Id: If1f520da6d9180862409af63296a1f90bad971e1

2014-12-31  Miklos Vajna  <vmiklos@collabora.co.uk>  [d6f6a4f6d7444177eaebc00198ed166f30540cbb]

	libvisio::XmlDrawingGenerator: handle user-defined metadata

	Change-Id: I63c4cf75009ae8303efd031c66570514180f2d9d

2014-12-31  Miklos Vajna  <vmiklos@collabora.co.uk>  [ce94c3eda0f65e965001fdb946b3db829fea2fbd]

	VSDXMetaData: missing breaks

	Change-Id: I6f1c1827761a532941fc699e912054966561978a

2014-12-30  Miklos Vajna  <vmiklos@collabora.co.uk>  [86ad7206b1d617817ac8699db740b613f3ed19a2]

	VSD: Visio doesn't write separate author and last modifier fields

	So author always equals to the last modifier. This is consistent with
	how the single timestamp is imported both as a creation and modification
	time, too.

	Change-Id: I65233316f412166dfd82c720c830cf625e1275b5

2014-12-30  Miklos Vajna  <vmiklos@collabora.co.uk>  [9cbe861b8f5e53a0aa83b863b1bf42224139dcfc]

	VSDX: import lastModifiedBy

	Change-Id: I5136c53cbca0b3dc1300e49bf82f290d8ff86ed8

2014-12-28  Miklos Vajna  <vmiklos@collabora.co.uk>  [d4684217ee9dd48e3352430e6d869824642f4082]

	VSD: import description

	Change-Id: I33b47f4ee903f9f4d07383df0552284de7401d7b

2014-12-28  Miklos Vajna  <vmiklos@collabora.co.uk>  [db443edcc59a9d48395e25f04a4c380d15877e4e]

	VSDX: import description

	Change-Id: I5e9e6fd793bb44c954e8255e70952badfb442d57

2014-12-28  Miklos Vajna  <vmiklos@collabora.co.uk>  [443227148dcbf4314369e1941e910f3b64290315]

	VSDXMetaData: check token type at a single place

	Change-Id: I0318d619a71b0785e2752a363a68f4db0b3acc0e

2014-12-28  Miklos Vajna  <vmiklos@collabora.co.uk>  [70e2cbc27a37d84c10e60438ed143a0cd391c796]

	VSD: import keywords

	Change-Id: I1896afdcbec2cb3177f053558b2a6668529bd85e

2014-12-27  Miklos Vajna  <vmiklos@collabora.co.uk>  [d3a5596c77afad6df538ad615111e0ee984b62e7]

	VSDX: import <cp:keywords>

	Change-Id: I7c61cde25389b5bee26885822d8995dfca7d031b

2014-12-26  Miklos Vajna  <vmiklos@collabora.co.uk>  [20434231174c9638c0c997199447ee638613a0a3]

	VSD: import author

	Change-Id: I3c73b61217ad8779ad9786acbe58ffada647ffbd

2014-12-25  Miklos Vajna  <vmiklos@collabora.co.uk>  [a3d033a7561cfca087d86632e2e3f0b215a1222e]

	VSDXMetaData: merge read*() methods into a single one

	Change-Id: I218bca925d5fc86d7b492e05b2bebbca2e5675d9

2014-12-24  Miklos Vajna  <vmiklos@collabora.co.uk>  [a110b6351ae38eccd0bd089918941fbfea677397]

	VSDX: import <dc:creator>

	Sigh, ODF has <dc:creator> as well, but with different semantics.

	Change-Id: I9425dc1b45c3c3cd9a1a97fb31f0ee087044814e

2014-12-23  Miklos Vajna  <vmiklos@collabora.co.uk>  [29dfeb9aee67f287a2c58c52084dca755e401df2]

	VSD: import subject

	Change-Id: I6b28cf0570ead1c7acf8b6b8354f23945ca0fdd0

2014-12-22  Miklos Vajna  <vmiklos@collabora.co.uk>  [e6a780adb90eb5122f03d4baeae3990972304bf5]

	VSDX: import <dc:subject>

	Change-Id: I8685be2883d3fa9719aafd0f1c1e928f863fb2ce

2014-12-21  Miklos Vajna  <vmiklos@collabora.co.uk>  [526368176b655a6f30d9f26b9cf3285367672b00]

	VSDXParser::processXmlNode: print debug messages to stderr

	Change-Id: I4d5b8414f9836dc75ef988985c64d929dbc6b336

2014-12-20  David Tardon  <dtardon@redhat.com>  [1ec587cab878f55b4552181f6a291e1b7c8d64e9]

	use the right form of c header

	Change-Id: I6cc11f41c84e47c1a49f5e553f1d067abd7ce79a

2014-12-20  Miklos Vajna  <vmiklos@collabora.co.uk>  [9f3264e596c9ffb9d690e69cfe8c86bc103885b1]

	VSD: test Creation/ModifiedTime import

	Change-Id: I5e0726cf8fecd7ec88beb4f338b3fad5c656fb5c

2014-12-17  Miklos Vajna  <vmiklos@collabora.co.uk>  [fa2420cddb9945caa7696cb394cec624f35c3782]

	ImportTest: fix indentation

	Change-Id: Ibc19c371d0aa8167193fcb3c3958caf2251abab9

2014-12-17  Miklos Vajna  <vmiklos@collabora.co.uk>  [c1ed56802e640b52674c62426b1e7507d09ddbf1]

	VSD: test metadata import

	Only title as a start.

	Change-Id: Ic7a4cc8ddd1f781466d7a113bf3a925ae112e874

2014-12-12  Miklos Vajna  <vmiklos@collabora.co.uk>  [c31e6cf3618f40ab4d1ec655e4e6f7c7590834ab]

	VSDX: test import of <dcterms:created> and <dcterms:modified>

	Change-Id: I78e6cc1811a2f0815f27d8780cf5f96208115b24

2014-12-10  Miklos Vajna  <vmiklos@collabora.co.uk>  [abdc998562acac7174ac8f83935404456109a141]

	fdo#86664 VSDX: test metadata import

	Only title as a start.

	Change-Id: I6d5f6e2dc1730ff2b2dfda760d575f605ab034eb

2014-12-08  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [109540e113867fd140838e5be79ca0e41c13d85c]

	Some Werror stuff fixing

	Change-Id: I6fbcf7168480590c1922e6f4ccd180e1605086dd

2014-12-07  Miklos Vajna  <vmiklos@collabora.co.uk>  [3507f0bc15a8f1c3d1c2b8d5d11962d8de5d9467]

	print debug messages to stderr

	Change-Id: I54305fcb8c2ea80951acdb5b426a0dd501330762

2014-12-05  David Tardon  <dtardon@redhat.com>  [25a2ba5fc9551f736192c3cca092aad69a98db09]

	Revert "avoid infinite recursion when reading a broken file"

	Streams can legitimately be read more than once. (In ideal world, we
	would keep the parsed data and just reuse it if/when the same stream
	pointer was encountered again, instead of parsing it repeatedly. There
	is a room for future improvement there :-)


src/src/lib/tokens.h
#ifndef __VSDXMLTOKENS_HXX__
#define __VSDXMLTOKENS_HXX__

const int XML_A = 1;
const int XML_ARAB = 2;
const int XML_ALIGNMENT = 3;
const int XML_ANGLE = 4;
const int XML_ARCTO = 5;
const int XML_B = 6;
const int XML_BENG = 7;
const int XML_BEGTRIGGER = 8;
const int XML_BEGINARROW = 9;
const int XML_BEGINX = 10;
const int XML_BEGINY = 11;
const int XML_BOTTOMMARGIN = 12;
const int XML_BULLET = 13;
const int XML_BULLETFONT = 14;
const int XML_BULLETFONTSIZE = 15;
const int XML_BULLETSTR = 16;
const int XML_C = 17;
const int XML_CANS = 18;
const int XML_CHER = 19;
const int XML_CASE = 20;
const int XML_CELL = 21;
const int XML_CHAR = 22;
const int XML_CHARACTER = 23;
const int XML_COLOR = 24;
const int XML_COLORENTRY = 25;
const int XML_COLORSCHEMEINDEX = 26;
const int XML_COLORS = 27;
const int XML_COMPANY = 28;
const int XML_CONNECTORSCHEMEINDEX = 29;
const int XML_D = 30;
const int XML_DEVA = 31;
const int XML_DBLUNDERLINE = 32;
const int XML_DEFAULTTABSTOP = 33;
const int XML_DOUBLESTRIKETHROUGH = 34;
const int XML_DRAWINGSCALE = 35;
const int XML_E = 36;
const int XML_ETHI = 37;
const int XML_EFFECTSCHEMEINDEX = 38;
const int XML_ELLIPSE = 39;
const int XML_ELLIPTICALARCTO = 40;
const int XML_EMBELLISHMENTINDEX = 41;
const int XML_ENDARROW = 42;
const int XML_ENDTRIGGER = 43;
const int XML_ENDX = 44;
const int XML_ENDY = 45;
const int XML_FACENAME = 46;
const int XML_FACENAMES = 47;
const int XML_FILL = 48;
const int XML_FILLBKGND = 49;
const int XML_FILLBKGNDTRANS = 50;
const int XML_FILLFOREGND = 51;
const int XML_FILLFOREGNDTRANS = 52;
const int XML_FILLPATTERN = 53;
const int XML_FLAGS = 54;
const int XML_FLIPX = 55;
const int XML_FLIPY = 56;
const int XML_FONT = 57;
const int XML_FONTSCALE = 58;
const int XML_FONTSCHEMEINDEX = 59;
const int XML_FOREIGN = 60;
const int XML_FOREIGNDATA = 61;
const int XML_GEOR = 62;
const int XML_GUJR = 63;
const int XML_GURU = 64;
const int XML_GEOM = 65;
const int XML_GEOMETRY = 66;
const int XML_HANG = 67;
const int XML_HANS = 68;
const int XML_HANT = 69;
const int XML_HEBR = 70;
const int XML_HEIGHT = 71;
const int XML_HIDETEXT = 72;
const int XML_HIGHLIGHT = 73;
const int XML_HORZALIGN = 74;
const int XML_IMGHEIGHT = 75;
const int XML_IMGOFFSETX = 76;
const int XML_IMGOFFSETY = 77;
const int XML_IMGWIDTH = 78;
const int XML_INDFIRST = 79;
const int XML_INDLEFT = 80;
const int XML_INDRIGHT = 81;
const int XML_INFINITELINE = 82;
const int XML_JPAN = 83;
const int XML_KHMR = 84;
const int XML_KNDA = 85;
const int XML_LAOO = 86;
const int XML_LAYER = 87;
const int XML_LAYERMEM = 88;
const int XML_LAYERMEMBER = 89;
const int XML_LEADER = 90;
const int XML_LEFTMARGIN = 91;
const int XML_LINE = 92;
const int XML_LINECAP = 93;
const int XML_LINECOLOR = 94;
const int XML_LINEPATTERN = 95;
const int XML_LINETO = 96;
const int XML_LINEWEIGHT = 97;
const int XML_LOCPINX = 98;
const int XML_LOCPINY = 99;
const int XML_MLYM = 100;
const int XML_MONG = 101;
const int XML_MASTER = 102;
const int XML_MASTERS = 103;
const int XML_MISC = 104;
const int XML_MOVETO = 105;
const int XML_NURBSTO = 106;
const int XML_NOFILL = 107;
const int XML_NOLINE = 108;
const int XML_NOSHOW = 109;
const int XML_ORYA = 110;
const int XML_OVERLINE = 111;
const int XML_PAGE = 112;
const int XML_PAGEHEIGHT = 113;
const int XML_PAGEPROPS = 114;
const int XML_PAGESCALE = 115;
const int XML_PAGESHEET = 116;
const int XML_PAGEWIDTH = 117;
const int XML_PAGES = 118;
const int XML_PARA = 119;
const int XML_PARAGRAPH = 120;
const int XML_PINX = 121;
const int XML_PINY = 122;
const int XML_POLYLINETO = 123;
const int XML_POS = 124;
const int XML_POSITION = 125;
const int XML_PRINT = 126;
const int XML_PROPERTIES = 127;
const int XML_QUICKSTYLEEFFECTSMATRIX = 128;
const int XML_QUICKSTYLEFILLCOLOR = 129;
const int XML_QUICKSTYLEFILLMATRIX = 130;
const int XML_QUICKSTYLEFONTMATRIX = 131;
const int XML_QUICKSTYLELINECOLOR = 132;
const int XML_QUICKSTYLELINEMATRIX = 133;
const int XML_QUICKSTYLESHADOWCOLOR = 134;
const int XML_QUICKSTYLETYPE = 135;
const int XML_QUICKSTYLEVARIATION = 136;
const int XML_REL = 137;
const int XML_RELCUBBEZTO = 138;
const int XML_RELELLIPTICALARCTO = 139;
const int XML_RELLINETO = 140;
const int XML_RELMOVETO = 141;
const int XML_RELQUADBEZTO = 142;
const int XML_RESIZEMODE = 143;
const int XML_RIGHTMARGIN = 144;
const int XML_ROUNDING = 145;
const int XML_ROW = 146;
const int XML_SINH = 147;
const int XML_SYRC = 148;
const int XML_SECTION = 149;
const int XML_SHAPE = 150;
const int XML_SHAPESHDWOFFSETX = 151;
const int XML_SHAPESHDWOFFSETY = 152;
const int XML_SHAPES = 153;
const int XML_SHDWBKGND = 154;
const int XML_SHDWFOREGND = 155;
const int XML_SHDWOFFSETX = 156;
const int XML_SHDWOFFSETY = 157;
const int XML_SHDWPATTERN = 158;
const int XML_SIZE = 159;
const int XML_SOLUTIONXML = 160;
const int XML_SPAFTER = 161;
const int XML_SPBEFORE = 162;
const int XML_SPLINE = 163;
const int XML_SPLINEKNOT = 164;
const int XML_SPLINESTART = 165;
const int XML_STRIKETHRU = 166;
const int XML_STYLE = 167;
const int XML_STYLESHEET = 168;
const int XML_STYLESHEETS = 169;
const int XML_TAML = 170;
const int XML_TELU = 171;
const int XML_THAA = 172;
const int XML_THAI = 173;
const int XML_TIBT = 174;
const int XML_TAB = 175;
const int XML_TABS = 176;
const int XML_TEMPLATE = 177;
const int XML_TEXT = 178;
const int XML_TEXTBKGND = 179;
const int XML_TEXTBKGNDTRANS = 180;
const int XML_TEXTBLOCK = 181;
const int XML_TEXTDIRECTION = 182;
const int XML_TEXTPOSAFTERBULLET = 183;
const int XML_TEXTXFORM = 184;
const int XML_THEMEINDEX = 185;
const int XML_TOPMARGIN = 186;
const int XML_TXTANGLE = 187;
const int XML_TXTHEIGHT = 188;
const int XML_TXTLOCPINX = 189;
const int XML_TXTLOCPINY = 190;
const int XML_TXTPINX = 191;
const int XML_TXTPINY = 192;
const int XML_TXTWIDTH = 193;
const int XML_UIGH = 194;
const int XML_VIET = 195;
const int XML_VARIATIONCOLORINDEX = 196;
const int XML_VARIATIONSTYLEINDEX = 197;
const int XML_VERTICALALIGN = 198;
const int XML_VISIBLE = 199;
const int XML_WIDTH = 200;
const int XML_X = 201;
const int XML_XFORM = 202;
const int XML_XFORM1D = 203;
const int XML_Y = 204;
const int XML_YIII = 205;
const int XML_A_ACCENT1 = 206;
const int XML_A_ACCENT2 = 207;
const int XML_A_ACCENT3 = 208;
const int XML_A_ACCENT4 = 209;
const int XML_A_ACCENT5 = 210;
const int XML_A_ACCENT6 = 211;
const int XML_A_ALPHA = 212;
const int XML_A_BGFILLSTYLELST = 213;
const int XML_A_CLRSCHEME = 214;
const int XML_A_CS = 215;
const int XML_A_DK1 = 216;
const int XML_A_DK2 = 217;
const int XML_A_EA = 218;
const int XML_A_EFFECTLST = 219;
const int XML_A_EFFECTSTYLE = 220;
const int XML_A_EFFECTSTYLELST = 221;
const int XML_A_EXT = 222;
const int XML_A_EXTLST = 223;
const int XML_A_EXTRACLRSCHEMELST = 224;
const int XML_A_FILLSTYLELST = 225;
const int XML_A_FMTSCHEME = 226;
const int XML_A_FOLHLINK = 227;
const int XML_A_FONT = 228;
const int XML_A_FONTSCHEME = 229;
const int XML_A_GRADFILL = 230;
const int XML_A_GS = 231;
const int XML_A_GSLST = 232;
const int XML_A_HLINK = 233;
const int XML_A_LATIN = 234;
const int XML_A_LIN = 235;
const int XML_A_LN = 236;
const int XML_A_LNSTYLELST = 237;
const int XML_A_LT1 = 238;
const int XML_A_LT2 = 239;
const int XML_A_LUMMOD = 240;
const int XML_A_MAJORFONT = 241;
const int XML_A_MINORFONT = 242;
const int XML_A_OBJECTDEFAULTS = 243;
const int XML_A_OUTERSHDW = 244;
const int XML_A_PRSTDASH = 245;
const int XML_A_SATMOD = 246;
const int XML_A_SCHEMECLR = 247;
const int XML_A_SHADE = 248;
const int XML_A_SOLIDFILL = 249;
const int XML_A_SRGBCLR = 250;
const int XML_A_SYSCLR = 251;
const int XML_A_THEME = 252;
const int XML_A_THEMEELEMENTS = 253;
const int XML_A_TINT = 254;
const int XML_CP = 255;
const int XML_CP_CATEGORY = 256;
const int XML_CP_COREPROPERTIES = 257;
const int XML_CP_KEYWORDS = 258;
const int XML_CP_LASTMODIFIEDBY = 259;
const int XML_DC_CREATOR = 260;
const int XML_DC_DESCRIPTION = 261;
const int XML_DC_LANGUAGE = 262;
const int XML_DC_SUBJECT = 263;
const int XML_DC_TEMPLATE = 264;
const int XML_DC_TITLE = 265;
const int XML_DCTERMS_CREATED = 266;
const int XML_DCTERMS_MODIFIED = 267;
const int XML_PP = 268;
const int XML_TP = 269;
const int XML_VT_BKGND = 270;
const int XML_VT_COLOR = 271;
const int XML_VT_CONNECTORFONTSTYLES = 272;
const int XML_VT_FILLPROPS = 273;
const int XML_VT_FILLSTYLES = 274;
const int XML_VT_FMTCONNECTORSCHEME = 275;
const int XML_VT_FMTCONNECTORSCHEMEEX = 276;
const int XML_VT_FMTCONNECTORSCHEMELINESTYLES = 277;
const int XML_VT_FMTSCHEMEEX = 278;
const int XML_VT_FMTSCHEMELINESTYLES = 279;
const int XML_VT_FONTPROPS = 280;
const int XML_VT_FONTSTYLES = 281;
const int XML_VT_FONTSTYLESGROUP = 282;
const int XML_VT_LINEEX = 283;
const int XML_VT_LINESTYLE = 284;
const int XML_VT_LINESTYLES = 285;
const int XML_VT_SCHEMEID = 286;
const int XML_VT_THEMESCHEME = 287;
const int XML_VT_VARCOLOR1 = 288;
const int XML_VT_VARCOLOR2 = 289;
const int XML_VT_VARCOLOR3 = 290;
const int XML_VT_VARCOLOR4 = 291;
const int XML_VT_VARCOLOR5 = 292;
const int XML_VT_VARCOLOR6 = 293;
const int XML_VT_VARCOLOR7 = 294;
const int XML_VT_VARSTYLE = 295;
const int XML_VT_VARIATIONCLRSCHEME = 296;
const int XML_VT_VARIATIONCLRSCHEMELST = 297;
const int XML_VT_VARIATIONSTYLESCHEME = 298;
const int XML_VT_VARIATIONSTYLESCHEMELST = 299;

const int XML_TOKEN_COUNT = 299;

const int XML_TOKEN_INVALID = -1;

#endif
